modules/refine-local-runner/src/test/java/org/openrefine/runners/local/pll/TextFilePLLTests.java

```diff
@@ -229,23 +239,132 @@ public void testReadIncompletePLL() throws IOException, InterruptedException {
-        truncateFile(new File(tempFile, "part-00003.gz"), 3);
-        truncateFile(new File(tempFile, "part-00004.gz"), 4);
-        truncateFile(new File(tempFile, "part-00005.gz"), 5);
-        truncateFile(new File(tempFile, "part-00006.gz"), 6);
-        truncateFile(new File(tempFile, "part-00007.gz"), 10);
-        truncateFile(new File(tempFile, "part-00008.gz"), 20);
+        truncateFile(new File(tempFile, "part-00001.zst"), 1);
+        truncateFile(new File(tempFile, "part-00002.zst"), 2);
+        truncateFile(new File(tempFile, "part-00003.zst"), 3);
+        truncateFile(new File(tempFile, "part-00004.zst"), 4);
+        truncateFile(new File(tempFile, "part-00005.zst"), 5);
+        truncateFile(new File(tempFile, "part-00006.zst"), 6);
+        truncateFile(new File(tempFile, "part-00007.zst"), 10);
+        truncateFile(new File(tempFile, "part-00008.zst"), 20);
         File successMarker = new File(tempFile, "_SUCCESS");
         successMarker.delete();
 
-        PLL<String> deserializedPLL = new TextFilePLL(context, tempFile.getAbsolutePath(), utf8, true);
-        Assert.assertEquals(deserializedPLL.getPartitions().size(), nbPartitions);
-        Assert.assertTrue(deserializedPLL.count() > 1000L); // it is 1196 as of 2023-06-02 with java 11
+        PLL<String> deserializedPLL = new TextFilePLL(context, tempFile.getAbsolutePath(), utf8, true, "end");
+        assertEquals(deserializedPLL.getPartitions().size(), nbPartitions);
+        assertTrue(deserializedPLL.count() > 1000L); // it is 1196 as of 2023-06-02 with java 11
         // but I suspect the exact count can be JVM-dependent
     }
 
+    @Test
+    public void testTouchAllPartitionsBeforeWriting() {
+        PLL<String> pll = parallelize(2, Arrays.asList("foo", "bar", "baz", "boo"))
+                .map(str -> {
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
+                        throw new IllegalStateException(e);
+                    }
+                    return str;
+                }, "artificially make PLL computation slow");
+
+        File tempFile = new File(tempDir, "touch_all_partitions_first");
+
+        // start saving the PLL to the disk
+        ProgressingFuture<Void> future = pll.saveAsTextFileAsync(tempFile.getAbsolutePath(), 0, false, false);
+        // immediately cancel the saving
+        future.cancel(false);
+
+        // the two files for each partition should be present already
+        File partition1 = new File(tempFile, "part-00000.zst");
+        assertTrue(partition1.exists());
+        File partition2 = new File(tempFile, "part-00001.zst");
+        assertTrue(partition2.exists());
+        File partition3 = new File(tempFile, "part-00002.zst");
+        assertFalse(partition3.exists());
+    }
+
+    @Test
+    public void testSynchronousRead() throws IOException, InterruptedException {
```

Comment:
This test demonstrates how a process may be able to read data from a PLL being written to the disk, in a synchronous way.